Directory.Build.props

```diff
@@ -119,6 +119,9 @@
 
       See Documentation/coding-guidelines/project-guidelines.md for more details on our configurations.
     -->
+    <!-- Hack to support .NET Framework debugging from VS. Configuration maps 1:1 to TargetGroup for netfx. -->
+    <TargetGroup Condition="'$(TargetGroup)' == '' AND ($(Configuration.StartsWith('net4')) OR $(Configuration.StartsWith('netfx')))">$(Configuration.Substring(0, $(Configuration.IndexOf('-'))))</TargetGroup>
```

Comment:
I'm not sure if I like this hack there are more then just VS scenarios that use Configuration property, for example Project-To-Project references we find the best configuration and then set Configuration property (see https://github.com/dotnet/corefx/blob/master/buildvertical.targets#L179). Putting this in here it seems like you are trying to reset the BuildConfiguration which would break some of those P2P scenarios.

Reply:
For Project-To-Project references is the supplied TargetGroup passed over or is it empty, like when launching from VS?

Reply:
It is empty and we parse the Configuration to set those properties but we don't override BuildConfiguration.

Reply:
My assumption was that if no TargetGroup but a Configuration is supplied we are launching from VS. It seems that is not right for the mentioned p2p references. Can we either pass a TargetGroup / the default TargetGroup in or another property that indicates that this is a p2p call and not a launch from VS?

I know this is hacky but I really want to make this work somehow as the benefits are nice.

Reply:
What happen to the idea of not depending on the BuildConfiguration? From your BuildTools PR it doesn't look like you are using that. What properties are incorrect that you need to override this?

Reply:
With whatever configuration passed in, `BuildingNETCoreAppVertical` always defaults to true. This is because we currently use the passed in TargetGroup to determine the vertical:

`<BuildingNETFxVertical Condition="'$(BuildingNETFxVertical)' == '' AND ('$(_bc_TargetGroup)'=='netfx' OR '$(BuildAllConfigurations)' == 'true')">true</BuildingNETFxVertical>`

Reply:
What do you need that property to be set for?

Reply:
for copying the right supplementals (config files, xunit assemblies), setting the right xunit arguments and referencing the right test sdk assemblies for Test Explorer work later.

Reply:
Unfortunately I think we need to find a better way to handle this then depending on the vertical build being set like this. This will cause unknown other affects. Perhaps we need to find better conditions or targets to cause those affects to happen.

For the xunit arguments aren't they static for a given lanuch setting if so why not just expand them out instead of depending on the properties, just like you are doing for the runner.

Reply:
> For the xunit arguments aren't they static for a given lanuch setting if so why not just expand them out instead of depending on the properties, just like you are doing for the runner.

Same arguments are conditionally via msbuild properties.

> Unfortunately I think we need to find a better way to handle this then depending on the vertical build being set like this. This will cause unknown other affects. Perhaps we need to find better conditions or targets to cause those affects to happen.

I'll think about it. Maybe conditioning the verticals differently is possible. Open to suggestions...

Reply:
>Some arguments are conditionally set via msbuild properties.

Can't you just hardcode the defaults? As people cannot set any extra msbuild properties differently from a VS build anyway.

Reply:
Of course that's possible. The main concern here are the xunit assemblies (with the runner) and config files which are needed to launch the runner: https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L131.

Reply:
From a quick look reading through commits like (https://github.com/dotnet/corefx/commit/664ca9508a03483a7fb2714b55b9cba885593d90) we need more then just the supplemental test data copied we also need the test binaries to be correctly bin-placed. I don't think hacking this the way you are doing will fix the issue. We might need to update our bin-placing and supplemental test data logic to copy to all applicable launch configurations if the configuration supports that configuration. @safern you have done some work in this area any thoughts?

Reply:
I agree with @weshaggard and we need to find a better way that doesn't depend on the vertical that is being built. Maybe we could just find a way to build the right binplace configuration and set binplace test to true so that the right assets are being copied? 

https://github.com/dotnet/corefx/blob/master/Directory.Build.targets#L38-L42

Would setting a private property and a flag to enable that binplace configuration work? And then adding that bin place configuration through that private property? Similar to _bc_TargetGroup?

Reply:
> We might need to update our bin-placing and supplemental test data logic to copy to all applicable launch configurations if the configuration supports that configuration. @safern you have done some work in this area any thoughts?

Do you mean that when we build the assembly we also want to prepare the test folders for all possible configurations? Will it still build all applicable configurations but use the supplied TargetGroup as the vertical when building from command line and passing a TargetGroup in? Also, we need to define what an applicable launch configuration is. If I haven't built a specific configuration from root already, i.e. `.\build -framework=netfx` this configuration shouldn't be built. Does that scenario also cover the cases where netstandard assemblies are only launched with the netcoreapp vertical? It would be great if we find a way to invoke them from within VS with a vertical different than netcoreapp.

Reply:
I'm imagining we would have a known set of platforms that we would support for launch settings, for now we could say netcoreapp and netfx. And if the vertical for that build has been done and an individual build was done that applies to one of the vertical builds it would bin-place for those.

Reply:
Sounds good. I'll take with @safern offline to explore the implementation options.